aidies-project/client/src/components/CampaignDetail.js

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useParams, useHistory } from 'react-router-dom';
+
+import { getOneCampaign, bid, setLocation } from '../store/campaign';
+import { DetermineBid, DetermineTimeRemaining } from './CampaignUtils';
+import { makeStyles } from '@material-ui/core/styles';
+import { Card, Typography, Button, Menu, MenuItem} from '@material-ui/core';
+import '../css/campaign.css';
+
+
+const useStyles = makeStyles({
+    details: {
+        display: 'flex',
+        flexDirection: 'column',
+    },
+    bullet: {
+        display: 'inline-block',
+        margin: '0 2px',
+        transform: 'scale(0.8)',
+    },
+    title: {
+        fontSize: 14,
+    },
+    pos: {
+        marginBottom: 12,
+    },
+    button: {
+        backgroundColor: '#222',
+        color: '#fff',
+    },
+    '&$focus': {
+        backgroundColor: 'green',
+    },
+    color: {
+        color: '#222',
+        
+    },
+    delete:{
+        color: '#FF0000'
+    }
+});
+const CampaignDetail = ({ needLogin}) => {
+    const classes = useStyles();
+    const {id} = useParams();
+    const history = useHistory();
+    const location = useLocation();
+    const campaign = useSelector(state => state.campaign.current);
+    let highest = useSelector(state => state.campaign.highestBid);
```

Comment:
How come you decided to use `let` for this variable?

Reply:
Leftover. Was accidentally trying to mutate the state. I switched this back to a `const`